src/main/java/frc/robot/Constants.java

```diff
@@ -5,132 +5,141 @@
-  public static double DRIVETRAIN_FRONT_RIGHT_ENCODER_OFFSET = (0.489502) * 2 * Math.PI; // 0.884521
-  public static double DRIVETRAIN_BACK_LEFT_ENCODER_OFFSET = (0.234375) * 2 * Math.PI; //  0.063965
-  public static double DRIVETRAIN_BACK_RIGHT_ENCODER_OFFSET = (0.520752) * 2 * Math.PI; // 0.811768
-
-  /*
-  ALL MOTOR GEAR RATIOS BELOW
-  */
-  public static final double DRIVE_MOTOR_GEAR_RATIO = 8.14; // this is the ratio of bot
-  public static final double ANGLE_MOTOR_GEAR_RATIO =
-      12.8; // https://www.swervedrivespecialties.com/products/mk4-swerve-module
-
-  public static final int BIG_ARM_GEAR_RATIO = 10;
-  public static final int SMALL_ARM_GEAR_RATIO = 7;
-
-  /*
-  MISC. NUMBERS BELOW
-  */
-  public static final double MAX_ANGLE_INACCURACY = Math.PI / 24;
-
-  public static final double BigArm_highTarget = 0.578857;
-  public static final double BigArm_mediumTarget = 0.437500;
-  public static final double BigArm_pickup = 0.392578;
-  public static final double BigArm_default = 0.386475;
-  public static final double BigArm_slider = 0.595092; // Fine tune
-  public static final double BigArm_pickupCone = 0.47295;
-
-  public static final double SmallArm_highTarget = -0.105736;
-  public static final double SmallArm_mediumTarget = 0.164062;
-  public static final double SmallArm_pickup = -0.014551;
-  public static final double SmallArm_default = 0.322266;
-  public static final double SmallArm_Slider = -0.18003028;
-  public static final double SmallArm_pickupCone = -0.272852;
-
-  // 15.479004 position for bigarm for smallarm to freely rotate
-  // 42.833496 highest position for big arm
-
-  // trackWidth - lateral distance between pairs of wheels on different sides of the robot
-  // wheelBase - distance between pairs of wheels on the same side of the robot
-  // THIS IS IMPORTANT FOR A RECTANGULAR ROBOT
-  public static final double TRACKWIDTH = 0.85;
-  public static final double WHEELBASE =
-      1.1333; // the ratio between the width and the length is around 3:4
-  // public static final double TRACKWIDTH = 1;
-  // public static final double WHEELBASE = 1;
-
-  public static final double SPEED_DIVISOR =
-      1; // what the max speed should be divided by, 1 is max power
-  public static final double ARM_SPEED = 0.005; // speed of the arms when adjusting manually
-
-  // CHARGE STATION COMMUNITY DISTANCE:
-  public static final double encoderToBlueChargeDistance = 96.4694981;
-  public static final double encoderToRedChargeDistance = 99;
-  // public static final double encoderToChargeDistance = 45.4694981;
-
-  public static final double encoderCenterCommunity = 100;
-  public static final double encoderToOutsideCommunityDistance = 87.30208217;
-  // 1.832716884 is the number of inches per 1 encoder value
-  // ~80 (plus offset) to the center of the charge station for robot
-  // ~160 is the distance to leave the community plus some extra cushion
+
+  public static final class CanCoderID {
```

Comment:
Can we nest the CAN IDs in another level? Something like the following?

```java
public final class Constants {
  public static final class CanId {
    public static final class CanCoder {
      public static final int FL = 43;
      public static final int FR = 44;
      public static final int BL = 42;
      public static final int BR = 41;
      public static final int SMALL_ARM = 22;
      public static final int BIG_ARM = 21;
    }
    public static final class Swerve {
      public static final class Angle {
        public static final int FL = 15;
        public static final int FR = 16;
        public static final int BL = 11;
        public static final int BR = 18;
      }
      public static final class Drive {
        public static final int FL = 17;
        public static final int FR = 12;
        public static final int BL = 13;
        public static final int BR = 14;
      }
    }
  }
  // ...
}
```

Then we should be able to reference them with `Constants.CanId.CanCoder.FL` and `Constants.CanId.Swerve.Angle.BR` and such